File: Data Structures & Algorithms/trapping-rain-water/submission-2.py
class Solution:
    def trap(self, height: List[int]) -> int:
        if not height:
            return 0
        l = 0
        r = len(height)-1
        LM,RM = height[l],height[r]
        res = 0
        while l<r:
            if LM < RM:
                l+=1
                LM = max(LM,height[l])
                res+= LM - height[l]
            else:
                r-=1
                RM = max(RM,height[r])
                res+= RM - height[r]
        return res

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = prices[0];
        int maxProfit = 0;

        for (int i = 1; i < prices.length; i++) {
            int profitIfSoldToday = prices[i] - minPrice;
            maxProfit = Math.max(maxProfit, profitIfSoldToday);
            minPrice = Math.min(minPrice, prices[i]);
        }

        return maxProfit;
    }
}
